tests/api/test_generated_endpoints.py

```diff
@@ -94,6 +95,45 @@ def test_get_data(self, generator, model, route, controlled_values, client, Grap
         for value in controlled_values:
             assert actual['data']['attributes'][value] == expected[camelCase_to_underscore(value)]
 
+    def test_can_delete_work(self, client, normalized_data_id):
+        graph = ChangeGraph([{
+            '@id': '_:abc',
+            '@type': 'workidentifier',
+            'uri': 'http://osf.io/faq',
+            'creative_work': {'@id': '_:789', '@type': 'dataset'}
+        }, {
+            '@id': '_:789',
+            'is_deleted': False,
+            '@type': 'dataset',
+        }])
+
+        graph.process()
+        preprint, identifier = ChangeSet.objects.from_graph(graph, normalized_data_id).accept()
+        assert preprint.is_deleted is False
+
+        encoded_id = IDObfuscator.encode(preprint)
+        response = client.get('/api/v2/{}/{}/'.format('datasets', encoded_id))
+        assert response.status_code == 200
+
+        graph = ChangeGraph([{
+            '@id': '_:abc',
+            '@type': 'workidentifier',
+            'uri': 'http://osf.io/faq',
+            'creative_work': {'@id': '_:789', '@type': 'dataset'}
+        }, {
+            '@id': '_:789',
+            'is_deleted': True,
+            '@type': 'dataset',
+        }])
+
+        graph.process()
+        ChangeSet.objects.from_graph(graph, normalized_data_id).accept()
+        preprint.refresh_from_db()
+        assert preprint.is_deleted is True
+
+        response = client.get('/api/v2/{}/{}/'.format('datasets', encoded_id))
+        assert response.status_code == 403
```

Comment:
You should also check the response body here to ensure that the error message is being sent back correctly and that you're encountering the correct error in the first place.